sdk/iot/Azure.Iot.Hub.Service/samples/IotHubClientSamples/MehodInvocationSamples.cs

```diff
@@ -0,0 +1,236 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading.Tasks;
+using Azure.Iot.Hub.Service.Models;
+
+namespace Azure.Iot.Hub.Service.Samples
+{
+    /// <summary>
+    /// This sample shows how to invoke a method on a device and module.
+    /// This sample requires the device sample to be running so that it can connect to the device.
+    /// </summary>
+    internal class MethodInvocationSamples
+    {
+        public readonly IotHubServiceClient IoTHubServiceClient;
+        public const int MaxRandomValue = 200;
+        public static readonly Random Random = new Random();
+
+        public MethodInvocationSamples(IotHubServiceClient client)
+        {
+            IoTHubServiceClient = client;
+        }
+
+        public async Task RunSampleAsync()
+        {
+            string deviceId = $"device{Random.Next(MaxRandomValue)}";
+            string moduleId = $"module{Random.Next(MaxRandomValue)}";
+
+            // Create a DeviceIdentity.
+            await CreateDeviceIdentityAsync(deviceId);
+
+            // Create a ModuleIdentity.
+            await CreateModuleIdentityAsync(deviceId, moduleId);
+
+            // Invoke method on device
+            await InvokeMethodOnDeviceAsync(deviceId);
```

Comment:
If you need the customer to start up another sample with the device side listening for method invocations, then you'll need some pause in this sample that says something like "Created device with id xxxxx, after starting the device client process, press any key to continue"

Reply:
Either that or don't create the device/module in this sample, and just have them be created before running the sample